src/main/scala/org/ergoplatform/db/ChangeSet.scala

```diff
@@ -0,0 +1,68 @@
+package org.ergoplatform.db
+
+import scorex.core.serialization.ScorexSerializer
+import scorex.util.serialization.{Reader, Writer}
+
+/**
+  * Describes a set of changes which could be reverted later.
+  * @param insertedKeys - inserted keys
+  * @param removed - removed entries (key, value)
+  * @param altered - altered keys (key, oldValue)
+  */
+final case class ChangeSet(insertedKeys: Seq[Array[Byte]],
+                           removed: Seq[(Array[Byte], Array[Byte])],
+                           altered: Seq[(Array[Byte], Array[Byte])])
+
+object ChangeSetSerializer extends ScorexSerializer[ChangeSet] {
+
+  override def serialize(obj: ChangeSet, w: Writer): Unit = {
+    w.putInt(obj.insertedKeys.size)
+    obj.insertedKeys.foreach { k =>
+      require(k.length <= 255, "Illegal key size")
+      w.putUByte(k.length)
+      w.putBytes(k)
+    }
+    w.putInt(x = obj.removed.size)
+    obj.removed.foreach { case (k, v) =>
+      require(k.length <= 255, "Illegal key size")
+      w.putUByte(k.length)
+      w.putBytes(k)
+      w.putInt(v.length)
+      w.putBytes(v)
+    }
+    w.putInt(obj.altered.size)
+    obj.altered.foreach { case (k, oldV) =>
+      require(k.length <= 255, "Illegal key size")
+      w.putUByte(k.length)
+      w.putBytes(k)
+      w.putInt(oldV.length)
+      w.putBytes(oldV)
+    }
+  }
+
+  override def parse(r: Reader): ChangeSet = {
+    val insertedQty = r.getInt()
+    val insertedKeys = (0 until insertedQty).foldLeft(Seq.empty[Array[Byte]]) { case (acc, _) =>
```

Comment:
`ChangeSet` is being deserialized only in case of rollback. So I would prefer to leave it written in convenient form, rather than rewriting it in imperative style.